### PR TITLE
feat: use search-service with dataservices search

### PR DESCRIPTION
**Pull request by geoffreyaldebert**
Issue https://github.com/datagouv/data.gouv.fr/issues/1405

Related to https://github.com/datagouv/udata-front/pull/565 and https://github.com/opendatateam/udata-search-service/pull/48

### CODE REVIEW

File: udata/core/dataservices/search.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            owner = User.objects(id=dataservice.owner.id).first()
+        extras = {}
+        for key, value in dataservice.extras.items():
+            extras[key] = to_iso_datetime(value) if isinstance(value, datetime.datetime) else value
+
+        return {
+            "id": str(dataservice.id),
+            "title": dataservice.title,
+            "description": dataservice.description,
+            "base_api_url	": dataservice.base_api_url,
```
Comment (magopian) [inline udata/core/dataservices/search.py:110]: is that a typo, having the extraneous spaces after `has_api_url`?

**Review by magopian — APPROVED** [review]
Nice! Good work!
A couple of nits and a possible typo?
It's fun to discover this work through the different PRs, and see it grow and evolve into the final feature, I feel like an investigator :D

File: udata/core/dataservices/search.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        "archived": BoolFilter(),
+    }
+
+    @classmethod
+    def is_indexable(cls, dataservice: Dataservice) -> bool:
+        return dataservice.deleted_at is None and not dataservice.private
+
+    @classmethod
+    def mongo_search(cls, args):
+        dataservices = Dataservice.objects(deleted_at=None, private__ne=True)
```
[the remark this answers is not in the document]
Comment (maudetes) [reply udata/core/dataservices/search.py]: :+1: good question, `visible` wasn't used either for Dataset, Reuse & Orga somehow. I've changed it in https://github.com/opendatateam/udata/pull/3171/commits/a3ec51a3e9e111e2d1b9b52f82aa2bd39948e0a0

File: udata/core/dataservices/search.py (hunk 1 above)
Comment (maudetes) [reply udata/core/dataservices/search.py]: :+1: fixed in https://github.com/opendatateam/udata/pull/3171/commits/c275f844d0ba32f142348a90b44182ef0986d7b9

File: udata/core/dataservices/search.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+        sort = (
+            cls.parse_sort(args["sort"])
+            or ("$text_score" if args["q"] else None)
+            or DEFAULT_SORTING
+        )
+        offset = (args["page"] - 1) * args["page_size"]
+        return dataservices.order_by(sort).skip(offset).limit(
+            args["page_size"]
+        ), dataservices.count()
```
[the remark this answers is not in the document]
Comment (maudetes) [reply udata/core/dataservices/search.py]: Indeed, trying it out in https://github.com/opendatateam/udata/pull/3171/commits/7634323c90d5220afa3854a034b536376cc1ec6b

**Pull request merged**